CONVENTIONS.md

```diff
@@ -0,0 +1,125 @@
+# Style conventions
+
+## Docstrings
+
+Docstrings shall be written in NumpyDoc format.
+
+Docstrings can either be just the summary text, or they must be complete.
+
+## Type annotations
+
+Type annotations will not be applied (for now).
+
+Note: I'm not very familiar with the latest and greatest type annotation
+options for libraries that make heavy use of numpy arrays like this one.
+
+## Linting
+
+Linting shall be performed with flake8, flake8-isort, flake8-black and pep8-naming.
+
+Automated formatting shall be performed with black and isort.
+
+Black is left at defaults, flake8 and isort are configured to adhere to black.
+
+## Testing
+
+All functions need to be covered by unit tests.
+
+
+# Functional API conventions
+
+This API is for internal use and for power-users that want to
+vectorize operations on large sets of things.
+
+Performance is prioritized over extensive input validation.
+
+The source for this API resides in the `pylinalg.func` subpackage and is organized
+by type.
+
+## Function naming
+
+The functional API has rather verbose names, but it makes things
+explicit.
+
+## Function signatures
+
+We strive to align closely with the conventions of [numpy](https://numpy.org/doc/stable/reference/ufuncs.html#ufuncs-kwargs)
+in order to be least-surprising for pylinalg users.
+
+All functions shall adhere as much possible to the following signature:
+
+`func(x1, x2, ..., /, *, out=None, dtype=None, **kwargs)`
+
+* All input arguments are positional-only.
+* There can be as many input arguments as needed for the function signature to make sense.
+* Input arguments are always [`array_like`](https://numpy.org/doc/stable/glossary.html?highlight=array_like#term-array_like):
+    * "Any scalar or sequence that can be interpreted as an ndarray. In
+      addition to ndarrays and scalars this category includes lists
+      (possibly nested and with different element types) and tuples.
+      Any argument accepted by `np.array` is `array_like`."
+    * TBD: require a call to `np.asarray` or `np.asanyarray` on inputs?
+* All non-input arguments are keyword-only.
```

Comment:
Ah, I see.  In light of this, we can either ignore my previous comment on keyword-only arguments or revise this statement. I think I still prefer my approach because it's a bit unclear to me what a non-input argument is.

Reply:
I revised it.